### DISCUSSION TITLE
Issue with Inconsistent Objects in Ray Tracing Experiment

### DISCUSSION THREAD

**Discussion by YuruZhang22** [NVlabs/sionna#490] [Q&A]
I am currently conducting a ray tracing experiment using the code provided below. However, I have encountered an issue where the objects in `paths_dict` differ with each execution of the code. My understanding is that these objects should represent the buildings that the ray passes through along a specific path.

Here are the main points I need assistance with:

1. **Inconsistency in Objects**: Why do the objects in `paths_dict` change with each run? 
2. **Selection of Objects**: Where in the code is the selection of these objects happening?
3. **Consistency of Variables**: Despite the change in objects, the variables `a` and `tau` within `cir` remain consistent. Why is this the case?

I am looking for a solution to ensure that the buildings (objects) that the ray tracing process passes through remain fixed across different runs. Any guidance or suggestions you can provide to address this issue would be greatly appreciated.

Below is the code I am using:

```py
scene = load_scene("/home/int/Desktop/UNL_eastcampus/east_campus.xml")

scene.tx_array = PlanarArray(num_rows=1,
                             num_cols=1,
                             vertical_spacing=0.5,
                             horizontal_spacing=0.5,
                             pattern="tr38901",
                             polarization="V")

scene.rx_array = PlanarArray(num_rows=1,
                             num_cols=1,
                             vertical_spacing=0.5,
                             horizontal_spacing=0.5,
                             pattern="iso",
                             polarization="V")

tx = Transmitter("tx", [-511.81,-367.08,34], [0.0,0.0,0.0])
scene.add(tx)
rx = Receiver("rx", [-420.57, -292.99, 1], [0.0,0.0,0.0])
scene.add(rx)

tx.look_at(rx)

paths = scene.compute_paths()

paths_dict = paths.to_dict()
for key, value in paths_dict.items():
    print(f"{key}: {value}")
```
Thank you for your time and assistance.

**Comment by SebastianCa** [comment]
Hi @YuruZhang22,
Is it possible that you are observing non-deterministic behavior due to the highly parallel code execution on GPUs? This can result in non-deterministic ordering of the paths.
